packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -168,7 +195,30 @@ export class AssociationManager {
     return `- Portfolio: ${portfolio.node.path} | Product: ${product.node.path}`;
   }
 
+  private static formatTemplateRule(scope: cdk.Stack, rule: TemplateRule): string {
+    return JSON.stringify({
+      [rule.ruleName]: {
+        Assertions: this.formatAssertions(scope, rule.assertions),
+        RuleCondition: rule.condition ? scope.resolve(rule.condition) : undefined,
+      },
+    });
+  }
+
+  private static formatAssertions(scope: cdk.Stack, assertions : TemplateRuleAssertion[]): {[key: string]: string | undefined }[] {
```

Comment:
```suggestion
  private static formatAssertions(stack: cdk.Stack, assertions : TemplateRuleAssertion[]): { [key: string]: string | undefined }[] {
```

Reply:
You can make the type here a lot more precise, no?

```suggestion
  private static formatAssertions(scope: cdk.Stack, assertions : TemplateRuleAssertion[]): { Assert: string,  AssertDescription: string | undefined }[] {
```